docs/cow-protocol/reference/contracts/core/README.mdx

```diff
@@ -0,0 +1,53 @@
+---
+sidebar_position: 1
+---
+
+# Core
+
+Core contracts of CoW Protocol are those that are necessary for the protocol to function.
+
+## Architecture
+
+The goal of the protocol is to bundle multiple user orders together to minimize fees deriving from using external liquidity sources and stave off maximal extractable value.
+
+User orders are signed off-chain and sent to the protocol. [Solvers compete in an auction](../core/auctions/rewards) to settle the orders. Orders without a direct match are settled by the solver using on-chain liquidity sources through the use of _interactions_.
+
+A settlement is a list of orders traded together, their prices, and the on-chain interactions necessary to retrieve external liquidity.
+
+To achieve this, the protocol needs to be able to:
+
+* Collect and verify user orders, interact with on-chain liquidity, store information on the state of the orders
+* Determine which addresses are solvers
+* Firewall user funds from solvers
+
+#### Guarantees and Invariants
+
+1. User funds are **NOT** accessible _except_ for settling an order authorized by the user
+2. A trade **MUST** be executed at the _limit price or better_ as specified in the order
+3. Once an order is fulfilled, it **CANNOT** be traded again
```

Comment:
Unfortunately this is technically untrue because of [this](https://github.com/cowprotocol/contracts/issues/29) security issue. I don't think we need to change the phrasing here, but we should probably point this out somewhere in the documentation and maybe here is the place.